### PR TITLE
Fix more API docs in core

### PR DESCRIPTION
**Pull request by pandamicro**
Re: https://github.com/cocos/cocos-engine/issues/10840

### Changelog

* API documentation improvements in core module

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by jk20012001 — APPROVED** [review]

File: cocos/core/components/renderable-component.ts [hunk 1]
```diff
@@ -41,11 +41,28 @@ const _matInsInfo: IMaterialInstanceInfo = {
...
+ * 默认情况下，渲染组件使用共享材质进行渲染，材质实例也不会被创建出来。仅在用户通过 [[material]]，[[materials]] 和 [[getMaterialInstance]] 接口获取材质时才会创建材质实例。
+ */
 @ccclass('cc.RenderableComponent')
 export class RenderableComponent extends Component {
-    @serializable
-    protected _visFlags = Layers.Enum.NONE;
-
+    /**
+     * @en The visibility which will be applied to the committed models.
+     * @zh 应用于所有提交渲染的 Model 的可见性
```
Comment (minggo) [inline cocos/core/components/renderable-component.ts:64]: Should add return type?

File: cocos/core/components/renderable-component.ts [hunk 2]
```diff
@@ -55,6 +72,18 @@ export class RenderableComponent extends Component {
         this._onVisibilityChange(val);
     }
 
+    /**
+     * @en Get the default shared material
+     * @zh 获取默认的共享材质
+     */
```
Comment (minggo) [inline cocos/core/components/renderable-component.ts:78]: The same as above.

File: cocos/core/components/renderable-component.ts [hunk 3]
```diff
@@ -77,13 +106,24 @@ export class RenderableComponent extends Component {
         }
     }
 
-    // _materials should be defined after sharedMaterials for Editor reset component reason
-    @type([Material])
-    protected _materials: (Material | null)[] = [];
+    /**
+     * @en The default material instance, it will create a new instance from the default shared material if not created yet.
+     * @zh 获取默认的材质实例，如果还没有创建，将会根据默认共享材质创建一个新的材质实例
+     */
```
Comment (minggo) [inline cocos/core/components/renderable-component.ts:112]: The same as above.

File: cocos/core/components/renderable-component.ts [hunk 4]
```diff
@@ -77,13 +106,24 @@ export class RenderableComponent extends Component {
...
+    get material (): MaterialInstance | null {
+        return this.getMaterialInstance(0);
+    }
+
+    set material (val: Material | MaterialInstance | null) {
```
Comment (minggo) [inline cocos/core/components/renderable-component.ts:117]: Does set and get share comments?

File: cocos/core/components/renderable-component.ts [hunk 5]
```diff
@@ -77,13 +106,24 @@ export class RenderableComponent extends Component {
...
+        }
+        this.setMaterialInstance(val, 0);
+    }
 
     /**
      * @en The materials of the model.
-     * @zh 模型材质。
+     * @zh 所有模型材质。
      */
     get materials (): (MaterialInstance | null)[] {
```
Comment (minggo) [inline cocos/core/components/renderable-component.ts:128]: Should add return type?

File: cocos/core/renderer/core/render-window.ts [hunk 6]
```diff
@@ -226,6 +229,12 @@ export class RenderWindow {
         }
     }
 
+    /**
+     * @en Extract all render cameras attached to the render window to the output cameras list
+     * @zh 将所有挂载到当前渲染窗口的摄像机存储到输出列表参数中
+     * @param cameras @en The output cameras list, should be empty before invoke this function
+     *                @zh 输出相机列表参数，传入时应该为空
```
Comment (minggo) [inline cocos/core/renderer/core/render-window.ts:236]: Can english and Chinese comments in different lines?

File: cocos/core/renderer/core/render-window.ts [hunk 7]
```diff
@@ -252,9 +261,9 @@ export class RenderWindow {
     }
 
     /**
-     * @zh
-     * 移除渲染相机
-     * @param camera 相机
+     * @en Detach a camera from the render window
+     * @zh 移除渲染相机
```
Comment (minggo) [inline cocos/core/renderer/core/render-window.ts]: 移除渲染相机 -> 移除场景的渲染县级？

File: cocos/core/root.ts [hunk 8]
```diff
@@ -589,6 +612,32 @@ export class Root {
         }
         l.destroy();
     }
+
+    private _init (): void {
```
Comment (minggo) [inline cocos/core/root.ts:616]: Why need this function?

File: cocos/core/root.ts [hunk 9]
```diff
@@ -589,6 +612,32 @@ export class Root {
...
+        if (JSB) {
+            this._naitveObj = new NativeRoot();
+        }
+    }
+
+    private _destroy (): void {
```
Comment (minggo) [inline cocos/core/root.ts:622]: The same as above.

File: cocos/core/root.ts [hunk 10]
```diff
@@ -589,6 +612,32 @@ export class Root {
...
+        if (JSB) {
+            this._naitveObj = null;
+        }
+    }
+
+    private _setCumulativeTime (deltaTime: number): void {
```
Comment (minggo) [inline cocos/core/root.ts:628]: The same as above.

File: cocos/core/root.ts [hunk 11]
```diff
@@ -589,6 +612,32 @@ export class Root {
...
+        this._cumulativeTime += deltaTime;
+        if (JSB) {
+            this._naitveObj.cumulativeTime = this._cumulativeTime;
+        }
+    }
+
+    private _setFrameTime (deltaTime: number): void {
```
Comment (minggo) [inline cocos/core/root.ts:635]: The same as above.

File: cocos/core/root.ts (hunk 8 above)
Comment (minggo) [reply cocos/core/root.ts:616]: Oh, it is in v3.5.2.

File: cocos/core/components/renderable-component.ts (hunk 4 above)
Comment (pandamicro) [reply cocos/core/components/renderable-component.ts:117]: Yes

File: cocos/core/renderer/core/render-window.ts (hunk 6 above)
Comment (pandamicro) [reply cocos/core/renderer/core/render-window.ts:236]: Yes

File: cocos/core/components/renderable-component.ts (hunk 1 above)
Comment (pandamicro) [reply cocos/core/components/renderable-component.ts:64]: It's a getter so no need to add return type, the property type will be automatically detected

File: cocos/core/components/renderable-component.ts (hunk 2 above)
Comment (pandamicro) [reply cocos/core/components/renderable-component.ts:78]: Same

File: cocos/core/components/renderable-component.ts (hunk 3 above)
Comment (pandamicro) [reply cocos/core/components/renderable-component.ts:112]: Same

File: cocos/core/renderer/core/render-window.ts (hunk 7 above)
Comment (pandamicro) [reply cocos/core/renderer/core/render-window.ts]: Fixed

**Review by minggo — APPROVED** [review]

**Pull request merged**